src/datasets/arrow_dataset.py

```diff
@@ -3594,14 +3634,18 @@ def push_to_hub(
         repo_id, split, uploaded_size, dataset_nbytes = self._push_parquet_shards_to_hub(
             repo_id=repo_id, split=split, private=private, token=token, branch=branch, shard_size=shard_size
         )
+
+        old_dataset_infos = self.get_remote_dataset_infos(repo_id=repo_id, token=token, branch=branch)
+
         organization, dataset_name = repo_id.split("/")
         info_to_dump = self.info.copy()
         info_to_dump.download_checksums = None
-        info_to_dump.download_size = uploaded_size
-        info_to_dump.dataset_size = dataset_nbytes
-        info_to_dump.size_in_bytes = uploaded_size + dataset_nbytes
+        info_to_dump.download_size = uploaded_size + old_dataset_infos.download_size
```

Comment:
I think we can add a bunch checks here. Typically if you upload a split with a different set of features, there's currently nothing preventing you from doing do, yet it'll crash when you use `load_dataset()`